src/libraries/Common/tests/System/Net/Prerequisites/NetCoreServer/Handlers/VerifyUploadHandler.cs

```diff
@@ -35,31 +35,40 @@ public static async Task InvokeAsync(HttpContext context)
             // Check MD5 checksum for non-empty request body.
             if (requestBodyBytes.Length > 0)
             {
-                // Get expected MD5 hash of request body.
-                string expectedHash = context.Request.Headers["Content-MD5"];
-                if (string.IsNullOrEmpty(expectedHash))
-                {
-                    context.Response.StatusCode = 400;
-                    context.Response.SetStatusDescription("Missing 'Content-MD5' request header");
-                    return;
-                }
-
-                // Compute MD5 hash of received request body.
-                string actualHash;
-                using (MD5 md5 = MD5.Create())
-                {
-                    byte[] hash = md5.ComputeHash(requestBodyBytes);
-                    actualHash = Convert.ToBase64String(hash);
-                }
-
-                if (expectedHash == actualHash)
+                string skip = context.Request.Headers["Content-MD5-Skip"];
+                if(!string.IsNullOrEmpty(skip)) 
                 {
                     context.Response.StatusCode = 200;
                 }
                 else
                 {
-                    context.Response.StatusCode = 400;
-                    context.Response.SetStatusDescription("Received request body fails MD5 checksum");
+                    // Get expected MD5 hash of request body.
+                    string expectedHash = context.Request.Headers["Content-MD5"];
+
+                    if (string.IsNullOrEmpty(expectedHash))
+                    {
+                        context.Response.StatusCode = 400;
+                        context.Response.SetStatusDescription("Missing 'Content-MD5' request header");
+                        return;
+                    }
+
+                    // Compute MD5 hash of received request body.
+                    string actualHash;
+                    using (MD5 md5 = MD5.Create())
+                    {
+                        byte[] hash = md5.ComputeHash(requestBodyBytes);
+                        actualHash = Convert.ToBase64String(hash);
+                    }
+
+                    if (expectedHash == actualHash)
+                    {
+                        context.Response.StatusCode = 200;
+                    }
+                    else
+                    {
+                        context.Response.StatusCode = 400;
+                        context.Response.SetStatusDescription("Received request body fails MD5 checksum");
+                    }
```

Comment:
Could we restrict this PR to just enabling tests that depend on the echo server, as the PR title says? If they need more changes, then that should go into a separate PR, so that it can get reviewed properly, IMHO.